### PR TITLE
Fix a couple of SF4 deprecations

### PR DESCRIPTION
**Pull request by lgeorget**
There are two deprecations left in the bundle that trigger under Symfony 4.4:

"Calling the "Symfony\Component\EventDispatcher\EventDispatcherInterface::dispatch()" method with the event name as first argument is deprecated since Symfony 4.3, pass it second and provide the event object first instead."

Events should depend on the abstraction Symfony\Contracts\EventDispatcher\Event, not the concretion Symfony\Component\EventDispatcher\Event.

### CODE REVIEW

**Review by Oliboy50 — DISMISSED** [review]
since this will drop support for older symfony versions, it would be best to make the whole bundle fully compatible with symfony `^4.4 || ^5.0` before bumping a new major release
(we still can merge this PR, but you may not see a new release before we support symfony 5)

**Comment by lgeorget** [comment]
Oops yes, Symfony 3.4 is still maintained (LTS version) so we cannot break it now (and probably shouldn't break it before some time) by merging this in the current major version. I'll check if it's possible to write the deprecations fix in such a way it keeps working for Symfony 3.4. Otherwise, I guess the way forward is, as you say, making a new major version for 4.4 || ^5.0 and leaving the current major version for 3.4 || ^4.0.

**Comment by lgeorget** [comment]
It works! The code is still compatible with Symfony 3.4 but no longer triggers the deprecations in Symfony 4.4. Now, it's your call to decide whether it's a good idea to use reflection to test the order of arguments in the event dispatcher and the existence of the Symfony\Contracts\EventDispatcher\Event class. It complicates everything. So maybe in the end, it would be better to have two releases, one for Symfony 3.4 (which is going to disappear soon-ish anyway) and the next one for Symfony 4.4 and 5.x.

Also, note that I had to exclude src/EventDispatcher/CassandraEvent.php from coke because of the double class definition in the file. If we manage to upgrade PHP_CodeSniffer, we could be more specific in the warnings we want to ignore (https://github.com/squizlabs/PHP_CodeSniffer/wiki/Advanced-Usage#ignoring-parts-of-a-file).

**Comment by lnahiro** [comment]
Thank you for your contribution, i close the PR because unfortunately this repo is not maintain anymore and will be archived.

**Comment by lgeorget** [comment]
Understood, we still use this bundle but we maintain our own fork now. We'll have to see how the upstream Cassandra C/C++ library evolves to see if it's worth it or if we should change solution altogether.

Anyway, thank you for the bundle for the time it was available best of luck in your new projects.

**Pull request closed without merging**
